bundler/lib/bundler/installer.rb

```diff
@@ -260,8 +260,8 @@ def resolve_if_needed(options)
       true
     end
 
-    def lock(opts = {})
-      @definition.lock(Bundler.default_lockfile, opts[:preserve_unknown_sections])
+    def lock
+      @definition.lock
```

Comment:
Why does the `Bundler::Runtime` preserve the opts, but this one doesn't?

Reply:
I removed `opts` because they were not being passed by any callers, so they felt misleading.

My understanding of why we're not passing `preserve_unknown_sections` here is the following:

* When `lock`ing from a `Bundler::Runtime` we never want lockfile updates, because that happens from `bundle exec` or a `bundler/setup` require and that causing lockfile modifications is generally unexpected. So we explicitly pass `preserve_unknown_sections: true`.

* When `lock`ing from `Bundler::Installer`, however, we want to respect [the rules defined in `Bundler::Definition`](https://github.com/rubygems/rubygems/blob/ed65279100234a17d65d71fe26de5083984ac5b8/bundler/lib/bundler/definition.rb#L329) for deciding whether to remove unknown sections or not.  So we don't pass anything.